plugins/woocommerce/src/Internal/Fulfillments/OrderFulfillmentsRestController.php

```diff
@@ -0,0 +1,1066 @@
+		}
+
+		// Return an error related to the request method.
+		$error_information = $this->get_authentication_error_by_method( $request->get_method() );
+
+		if ( is_null( $error_information ) ) {
+			return false;
+		}
+
+		return new \WP_Error(
+			$error_information['code'],
+			$error_information['message'],
+			array( 'status' => rest_authorization_required_code() )
+		);
+	}
+
+	/**
+	 * Get the fulfillments for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The fulfillments for the order, or an error if the request fails.
+	 */
+	public function get_fulfillments( WP_REST_Request $request ): WP_REST_Response {
+		$order_id     = (int) $request->get_param( 'order_id' );
+		$fulfillments = array();
+
+		// Fetch fulfillments for the order.
+		try {
+			$datastore    = wc_get_container()->get( FulfillmentsDataStore::class );
+			$fulfillments = $datastore->read_fulfillments( WC_Order::class, "$order_id" );
+		} catch ( \Exception $e ) {
+			return $this->prepare_error_response(
+				$e->getCode(),
+				$e->getMessage(),
+				WP_Http::BAD_REQUEST
+			);
+		}
+
+		// Return the fulfillments.
+		return new WP_REST_Response(
+			array(
+				'fulfillments' => array_map(
+					function ( $fulfillment ) {
+						return $fulfillment->get_raw_data(); },
+					$fulfillments
+				),
+			),
+			WP_Http::OK
+		);
+	}
+
+	/**
+	 * Create a new fulfillment with the given data for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The created fulfillment, or an error if the request fails.
+	 */
+	public function create_fulfillment( WP_REST_Request $request ) {
+		$order_id = (int) $request->get_param( 'order_id' );
+
+		// Create a new fulfillment.
+		try {
+			$fulfillment = new Fulfillment();
+			$fulfillment->set_props( $request->get_json_params() );
+			$fulfillment->set_meta_data( $request->get_json_params()['meta_data'] );
+			$fulfillment->set_entity_type( WC_Order::class );
+			$fulfillment->set_entity_id( "$order_id" );
+
+			$fulfillment->save();
+		} catch ( \Exception $e ) {
+			return $this->prepare_error_response(
+				$e->getCode(),
+				$e->getMessage(),
+				WP_Http::BAD_REQUEST
+			);
+		}
+
+		return new WP_REST_Response( array( 'fulfillment' => $fulfillment->get_raw_data() ), WP_Http::CREATED );
+	}
+
+	/**
+	 * Get a specific fulfillment for the order.
+	 *
+	 * @param WP_REST_Request $request The request object.
+	 *
+	 * @return WP_REST_Response The fulfillment for the order, or an error if the request fails.
+	 */
+	public function get_fulfillment( WP_REST_Request $request ): WP_REST_Response {
+		$order_id       = (int) $request->get_param( 'order_id' );
+		$fulfillment_id = (int) $request->get_param( 'fulfillment_id' );
+
+		// Fetch the fulfillment for the order.
+		try {
+			$fulfillment = new Fulfillment( $fulfillment_id );
+			$this->validate_fulfillment( $fulfillment, $fulfillment_id, $order_id );
```

Comment:
@tpaksu, we need to handle the return value of the `validate_fulfillment`, or we are allowing the fulfillment ID to be retrieved from any order, even if it's not its parent order (the `woocommerce_rest_fulfillment_invalid_id` response is never returned).
One option would be to only encapsulate the condition from the `if`, sorry, I was not very clear with my previous comment around it.
```php
if ( ! $this->validate_fulfillment( $fulfillment, $fulfillment_id, $order_id ) ) {
	return $this->prepare_error_response(
		'woocommerce_rest_fulfillment_invalid_id',
		__( 'Invalid fulfillment ID.', 'woocommerce' ),
		WP_Http::NOT_FOUND
	);
}
```
Does this make sense? 🤔

Reply:
Opps, I thought it was throwing an exception! I'll fix it ASAP.